### PR TITLE
training with moreh

### PR DESCRIPTION
**Pull request by huonglarne**
- mlflow in training script
- dependencies

### CODE REVIEW

File: trl/trainer/moreh_utils.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            if 'loss' in state.log_history[-1]:
+                state.log_history[-1]["throughput"] = throughput
+                state.log_history[-1]["step"] = state.global_step
+
+                mlflow.log_metric("lr", state.log_history[-1]["learning_rate"] , step=state.global_step)
+                mlflow.log_metric("throughput", throughput , step=state.global_step)
+                state.log_history[-1]["step"] = state.global_step
+
+                mlflow.log_metric("lr", state.log_history[-1]["learning_rate"] , step=state.global_step)
+                mlflow.log_metric("throughput", throughput , step=state.global_step)
```
Comment (yeomni) [inline trl/trainer/moreh_utils.py:37]: Does it have to have two redundant log_metric lines for lr and throughput?

File: trl/trainer/moreh_utils.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+                print(f'loss: {state.log_history[-1]["loss"]}, lr: {state.log_history[-1]["learning_rate"]}, throughput: {throughput}, step: {state.global_step}')
```
Comment (yeomni) [inline trl/trainer/moreh_utils.py:39]: `loss` should also be logged to mlflow.log_metric

**Review by yeomni — CHANGES_REQUESTED** [review]
Except for these two comments, LGTM.

**Review by yeomni — APPROVED** [review]
